functions_complete.py

```diff
@@ -0,0 +1,310 @@
+        while is_position_valid == False: # until is_position_valid is set to True, this loop continues
+            orientation, y_start_position, x_start_position = generate_random_position(size_ship)
+            is_position_valid = check_position(grid, size_ship, orientation, y_start_position, x_start_position)
+        
+        # Once the position is valid, the grid is filled with the current ship
+        if (orientation == 0):
+            y_to_fill = slice((y_start_position), (y_start_position + size_ship), 1)
+            x_to_fill = x_start_position
+        elif (orientation == 1):
+            y_to_fill = y_start_position
+            x_to_fill = slice((x_start_position), (x_start_position + size_ship), 1)
+        
+        grid[y_to_fill, x_to_fill] = 1
+        # Is there a way to name the ship that I'm currently using?
+        print("The", ships[ship], "was just placed")
+        print("Current grid:")
+        print(grid)
+    
+    # Now remove the first and last columns and rows because they were only included to allow the check_position function
+    grid = grid[1:11, 1:11]
+    print("Final grid:")
+    print(grid)
+    
+    return (grid)
+
+######## Then create the player's grid
+def position_players_ships(ships, size_ships):
+    """
+    This function should position the ships that the player chooses to the player's grid.
+
+    Parameters
+    ----------
+    ships : TYPE
+        List of the ships to be positioned.
+    size_ships : TYPE
+        List of the size of the ships to be positioned.
+
+    Returns
+    -------
+    grid : TYPE
+        The grid with the ships that the player positioned.
+
+    """
+    grid = np.zeros((12, 12))
+    for ship in range(0, len(size_ships)):
+        size_ship = size_ships[ship]
+        print("You now have to place the", ships[ship], "which has the size", size_ship)
+        is_position_valid = False
+        while is_position_valid == False: # until is_position_valid is set to True, this loop continues
+            orientation = int(input("Enter which orientation you want your ship to have (0 is vertical and 1 is horizontal): "))
+            y_start_position = int(input("Enter at which row you want your ship to start (a value between 1 and 10):"))
+            x_start_position = int(input("Enter at which column you want your ship to start (a value between 1 and 10):"))
+            is_position_valid = check_position(grid, size_ship, orientation, y_start_position, x_start_position)
+            print("You cannot place this ship here. Enter a new position")
+        
+        # Once the position is valid, the grid is filled with the current ship
+        if (orientation == 0):
+            y_to_fill = slice((y_start_position), (y_start_position + size_ship), 1)
+            x_to_fill = x_start_position
+        elif (orientation == 1):
+            y_to_fill = y_start_position
+            x_to_fill = slice((x_start_position), (x_start_position + size_ship), 1)
```

Comment:
I believe you had something very similar in one of the previous functions. Maybe it is worth taking these operations in a new function that you can just add whenever you need to fill the grid. It can shorten your code if this operation is necessary to execute a few times.